[21기 박현찬] Modify : 리뷰 피드백 반영

Eyes0n opened the pull request:
## 수정 사항 간략한 한줄 요약
- Login | 내부 라우팅에 대한 a태그 -> Link태그로 변환 
- Main | main컴포넌트 story부분의 Image src를 인라인 스타일에서 scss파일에서 경로 지정
- Main | 비구조화 할당 미흡한 부분 적용
		 
## 수정 사항들 자세한 내용
- 내부 라우팅은 Link태그, 외부 라우팅은 a 태그로 용도에 맞게 적용
- scss파일에서 public 하위에 있는 이미지 파일을 불러오기 위해 scss파일 안 해당 이미지 태그의 경로에 `'http://localhost:3000`을 포함한 경로로 명시해줘야한다. 
ex) `url('http://localhost:3000/images/hyunchanpark/images/insta-sprite.png') `
- 비구조화 할당 적용

## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Eyes0n committed:
Add : Common component인 Nav 반영 완료

Eyes0n committed:
Add : 버튼 활성화 및 로그인 유효성 검사 기능 추가
- input 입력 값에 대한 유효성 검사 using 정규표현식
- 로그인 버튼 활성화 기능 구현 완료
- react-router의 withRouter를 사용하여 submit 이벤트 시 main으로 라우팅

Eyes0n committed:
Modify : Main과 Login 관련 scss 용도별 파일 분리 및 적용
- 각 scss파일에 있던 변수와 mixin을 독립적인 파일로 분리 -> _variable.scss & _mixin.scss
- common.scss에 import하여 스타일 적용

Eyes0n committed:
Add : Misson 1 & 2 기능 구현 완료
- input태그에 name attribute를 부여하여 1개  handleInput으로 2개의 input관리
- regExp를 사용하여 id와pw의 유효성 검증 -> 로그인 버튼 활성화

Eyes0n committed:
Add : Misson 3 & 4 기능 구현 완료
- 댓글 영역 comment.js 파일로 컴포넌트 분리
- Main.js 파일에서 map을 사용하여 commet컴포넌트 출력
- comment컴포넌트에 props로 각 comment에 대한 정보 전달
- 각 comment에 대해 사용자 정보에 따른 삭제버튼 출력 유무 및 삭제 기능 구현 완료
- 좋아요 기능 구현 필요

Eyes0n committed:
Modify : 로그인 페이지 라우팅, scss파일
- 로그인에서 메인으로 라우팅 기능 함수 잘못된 코드 수정
- varialbe와 mixin 용도에 맞게 다시 분리

Eyes0n committed:
Add : Mission 1) commentData.js로 구현

Eyes0n committed:
Add : Mission 1) commentData.json 으로 구현

Eyes0n committed:
Add : Mission 2) mock data로 여러 개의 피드 구현 완료
- main에서 feeds & feed 컴포넌트 분리

Eyes0n committed:
Refactoring

fghjjkl32 commented on src/Components/Nav/Nav.js:
```diff
@@ -61,7 +61,10 @@ class Nav extends React.Component {
             </a>
             <a href="/">
```
a링크 사용하신게 많이 보입니다!
a 링크대신 Link를 사용하는게 좋아보입니다!

yeonuk-hwang replied:
[참고](https://stackoverflow.com/questions/43087007/react-link-vs-a-tag-and-arrow-function)

`<a>`는 사용하면 새로고침 하는 것처럼 html을 새로 다 받아 오는 반면, `<Link>`를 사용하면 컴포넌트만 바꿔줍니다. 렌더링 최적화를 위해서 `<Link>` 사용해주세요!

Eyes0n replied:
리뷰 반영했습니다

Eyes0n replied:
리뷰 반영했습니다.

Eyes0n replied:
리뷰 반영했습니다. 대부분 a태그 -> Link태그로 수정 하지만 Login페이지의 가입하기 a태그는 인스타 가입하기 링크로 적어놔서 바꾸지 않았습니다.

fghjjkl32 commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+      this.validate(pw, regObj.passwordRegExp)
+    );
+  };
+
```
정규식 표현을 사용해서 이런식으로 테스트 돌릴 수 있군요 참고하겠습니다 !! 👍

yeonuk-hwang replied:
기능별 함수 분리하고 잘 재활용하신 것 같습니다!👍

fghjjkl32 commented on public/data/feedData.json:
```diff
@@ -0,0 +1,52 @@
+      {
+        "id": 3,
+        "userName": "jayPark",
+        "content": "Hey.",
```
json 데이터를 다양하게 활용하셔서 보기 좋습니다!

fghjjkl32 commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+                      type="text"
+                      name="userId"
+                      placeholder="전화번호, 사용자 이름 또는 이메일"
+                      value={this.state.userId}
+                      onChange={this.handleInput}
```
비구조화 할당을 따로 해주셨는데 사용을 안하고 계십니다!
함수도 비구조화 할당 가능해보입니다 !

yeonuk-hwang replied:
건우님이 리뷰 잘 남겨주셨네요 확인하고 반영해보세요!👍

Eyes0n replied:
리뷰 반영했습니다.

Eyes0n replied:
리뷰 반영했습니다.

fghjjkl32 commented on src/Pages/hyunchanpark/Main/Comment.js:
```diff
@@ -0,0 +1,51 @@
+
+class Comment extends React.Component {
+  render() {
+    const {
+      id,
+      userName,
+      commentValue,
+      defaultComments,
+      isLiked,
+      onToggleLike,
+      onRemoveComment,
+    } = this.props;
```
비구조화 할당 보기 좋습니다 👍

fghjjkl32 commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,161 @@
+    };
+  }
+
+  // componentDidMount() {
+  //   this.setState({
+  //     commentList: COMMENT,
+  //   });
+  // }
```
사용을 안하시는 코드면 삭제를 해주시는게 좋아 보입니다!

Eyes0n replied:
리뷰 반영했습니다.

fghjjkl32 commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,161 @@
+    });
+  };
+
+  onToggleLike = e => {
+    const { commentList } = this.state;
+    console.log(e.target.id);
+    this.setState({
+      isLiked: !commentList[e.target.id].isLiked,
+    });
+  };
```
좋아요 까지 구현하신건가요 🔥

Eyes0n replied:
아니요......삭제까지 구현했습니다. ㅠㅠ

fghjjkl32 commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,161 @@
+    const { commentList, commentValue } = this.state;
+    if (commentValue === '') return;
+
+    this.setState({
+      commentList: [
+        ...commentList,
+        {
+          id: commentList.length + 1,
+          userName: 'hyunchan',
+          content: commentValue.trim(),
+          isLiked: false,
+        },
+      ],
+      commentValue: '',
+    });
+  };
```
json데이터를 복사해서 이렇게 사용도 가능하군요 참고하겠습니다~

fghjjkl32 commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,161 @@
+          </div>
+
+          <div className="other-comments">
+            {commentList.map((comment, i) => (
+              <Comment
+                key={comment.id}
+                id={comment.id}
```
key따로 id따로 데이터를 전달해주신거 너무 좋은거 같네요! 👏🏻

fghjjkl32 commented on src/Pages/hyunchanpark/Main/Feed.js:
```diff
@@ -0,0 +1,151 @@
+                  </a>
+                  <span className="city">
+                    {city}
+                    {/* <span>•</span>
+                    195 */}
```
사용하지 않는 console.log 발견~ 🛸

fghjjkl32 commented on src/Pages/hyunchanpark/Login/Login.scss:
```diff
@@ -0,0 +1,160 @@
+  @include flex-set(row, center, center);
+  width: 100%;
+  height: 100vh;
+  background-color: $background-default-color;
+
+  .entire {
+    width: 350px;
+    background-color: $background-default-color;
+
+    .entire_top {
+      height: 380px;
+
+      &.content {
+        padding: 10px 0;
+        margin: 0 0 10px;
+        border: $border-default-set;
+
+        .login_container {
+          @include flex-set(column, center, center);
+          width: 100%;
```
sass nesting 사용 깔끔, 간결 가독성 👍

fghjjkl32 commented on src/Pages/hyunchanpark/Main/Feed.js:
```diff
@@ -0,0 +1,151 @@
+                    <svg
+                      aria-label="좋아요"
+                      className="_8-yf5"
+                      fill={feedLike ? 'red' : '#262626'}
```
Boolean 데이터 타입의 활용을 하나하나 꼼꼼하게 자주 하신점 👏🏻 박수~~~

fghjjkl32 commented on src/Pages/hyunchanpark/Main/Main.js:
```diff
@@ -0,0 +1,246 @@
+                        <a href="/" className="story-link">
+                          <div
+                            className="link-background"
+                            style={{
+                              background:
+                                'url("/images/hyunchanpark/images/insta-sprite.png) no-repeat -306px -66px',
+                            }}
```
리액트 인라인 스타일로 꾸미는법 까먹고 있었는데 참고 하겠습니다~

fghjjkl32 commented on src/Pages/hyunchanpark/Main/Main.js:
```diff
@@ -0,0 +1,246 @@
+                            />
+                          </a>
+                          <div className="username">
+                            <a href="/">imsj208</a>
+                            <span>kingwoobinkim님 외 6명이 팔로우합니다</span>
+                          </div>
+                        </div>
+                        <button className="follow-button transparent">
+                          팔로우
+                        </button>
+                      </li>
+                      <li className="recommend-item">
+                        <div className="item">
+                          <a className="item-link" href="/">
+                            <img
+                              src="/images/hyunchanpark/images/insta-sprite.png"
+                              alt="other user"
```
요기 코드도 전체로 a링크를 주셨습니다 Link로 주시면 좋을거 같습니다

Eyes0n replied:
리뷰 반영했습니다.

fghjjkl32 commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,161 @@
+
+        <div className="post-comment">
+          <div className="comment-form-container">
+            <form className="comment-form" onSubmit={this.onSubmitCommentForm}>
+              <button className="emoji-button">
```
form태그의 submit을 추가하셔서 엔터키를 쳐도 입력이 가능하게 한점 보기 좋습니다!
이렇게 하면 따로 엔터키 함수를 만들지 않아도 되니까 좋은거 같습니다 !

fghjjkl32 reviewed (commented):
제가 감히 리뷰를 진행하도록 하겠습니다..! 코드가 전체적으로 깔끔한거 같습니다 💯 🥇

Eyes0n committed:
Modify : commentList에 있던 함수 및 스테이트 FeddList로 올김
- 컴포넌트 분리 feedList -> feed -> commentList & commentInput -> comment

Eyes0n committed:
Add : 로그인/회원가입 요청 fetch 적용
- 회원가입/로그인 id와 pw의 validate에 따른 성공/실패/오류 메시지 확인
- 로그인 성공 시 JWT를 로컬스토리지에 저장 후 통신마다 사용

Eyes0n committed:
Modify : 컴포넌트와 state의 구조 개편 & 댓글 추가/삭제/좋아요/좋아요 취소 기능 추가
- 하위 컴포넌트에서 관리되던 state들을 FeedList에서 feedList state로 통합하여 중앙 관리
- inputComment컴포넌트만 독립적인 state를 가지게 하여 랜더링 영역을 분할시킴
- inputComment의 state값을 함수이 인자로 FeedList까지 전달시켜 결국 feedList에 반영하게 만듬
- 댓글 추가/삭제 기능 추가
- 댓글 좋아요/좋아요 취소 기능 추가

Eyes0n committed:
Add : state 중앙 관리 최종 버전 & 댓글 관련 기능 추가
- 하위 컴포넌트에서 관리되던 state들을 FeedList에서 feedList state로 통합하여 중앙 관리
- inputComment컴포넌트만 독립적인 state를 가지게 하여 랜더링 영역을 분할시킴
- inputComment의 state값을 함수이 인자로 FeedList까지 전달시켜 결국 feedList에 반영하게 만듬
- 댓글 추가/삭제 기능 추가
- 댓글 좋아요/좋아요 취소 기능 추가

yeonuk-hwang commented on src/Components/Nav/Nav.scss:
```diff
@@ -22,6 +22,7 @@ nav {
     .title {
       margin: 0;
       font-size: 30px;
+      font-weight: 400;
```
font-weight:400이 기본속성인데 따로 부여하신 이유가 있나요?

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+import { withRouter } from 'react-router';
+import './Login.scss';
+
+const regObj = {
+  emailRegExp:
+    /[a-zA-Z0-9.-_+!]+@[a-zA-Z0-9]+\.[a-zA-Z0-9]{2,}(?:.[a-zA-Z0-9]{2,3})?/,
+  passwordRegExp: /[a-zA-Z0-9]{5,100}/,
+};
```
이 값은 컴포넌트내부에 값으로 가지고 있는게 좋아보이는데 컴포넌트 바깥에서 선언하신 이유가 있나요?

Eyes0n replied:
상수로 사용하기 위해서 class 밖에 선언해 놨는데 컨벤션에 맞지 않네요 ㅎㅎ
REGEXP로 변경했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+
+  validate = (value, regExp) => {
+    const reg = new RegExp(regExp);
+    let isValid = false;
+    if (reg.test(value)) isValid = true;
+    return isValid;
```
```suggestion
return reg.test(value)
```
test method의 return값 자체가 boolean이기 때문에 위와 같이 간결하게 표현 가능해보입니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+        [e.target.name]: e.target.value,
+      },
+      () => {
+        this.validateInputData(this.state.userId, this.state.userPw);
```
validateInputData 를 호출하고 계시지만 이 함수는 실행되기만 하고 리턴값을 어디에도 활용하지 않고 있는 것 같습니다!
실질적으로 어떤 의미도 없는 코드같습니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+  handleInput = e => {
+    this.setState(
+      {
+        [e.target.name]: e.target.value,
```
`const {name, value} = e.target;` 으로 구조분해해서 사용하시면 좀 더 간결하겠네요!

Eyes0n replied:
이렇게 한 줄로 간결하게 쓸 수 있는 코드를 구조 분해 할당으로 2줄로 만들어 줘야하는 이유를 모르겠습니다.
다만, 현재 함수 구현하면서 많이 쓰이는 state & props 에 한해 구조 분해 할당을 해주고 있습니다

하지만 리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+  handleSubmit = e => {
+    e.preventDefault();
+
+    if (!this.validateInputData(this.state.userId, this.state.userPw)) return;
```
👍
state값도 구조분해해서 사용하면 좀 더 간결해보일 것 같습니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+
+  render() {
+    const { userId, userPw } = this.state;
+    const isDisabled = this.validateInputData(userId, userPw) ? false : true;
```
애초에 boolean을 리턴하는 함수이기 때문에 not 연산자를 활용해서
```suggestion
    const isDisabled = !this.validateInputData(userId, userPw)
```
button tag에서 한번만 사용되는 값이라면 따로 변수선언 할당과정 거칠 필요 없이 버튼 태그 내에 바로 적어줘도 될 것 같습니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.js:
```diff
@@ -0,0 +1,127 @@
+                      <button
+                        className="login-button"
+                        type="submit"
+                        disabled={isDisabled}
```
```suggestion
                        disabled={!this.validateInputData(userId, userPw)}
```
위에 달아드린 코멘트와 이어지는 내용입니다!

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.scss:
```diff
@@ -0,0 +1,160 @@
+@import '../Styles/common.scss';
+
+button {
```
컴포넌트 이름으로 nesting 적용해주세요!
이렇게 되면 다른 컴포넌트들의 스타일에도 영향을 미칠 수 있습니다!

yeonuk-hwang replied:
공통적으로 적용되어야 하는 속성이라면 common에 넣어주시는 게 맞습니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.scss:
```diff
@@ -0,0 +1,160 @@
+        border: $border-default-set;
+
+        .login_container {
+          @include flex-set(column, center, center);
```
mixin 잘 활용해주셨네요!

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.scss:
```diff
@@ -0,0 +1,160 @@
+
+        .login_container {
+          @include flex-set(column, center, center);
+          width: 100%;
```
만약 애초에 display:block 속성이라면 width:100%를 부여할 필요가 없어보입니다!
한번 확인해보세요!

Eyes0n replied:
css -> scss 로 변환 작업할 때 block임에도 불구하고 width가 100%가 default로 적용되지 않아 '또는' 글씨 양쪽 가로 막대에 flex-grow:1 이 안먹혀서 width값을 명시해줬는데
지금 지우니 잘 적용되네요 제 불찰입니다.. 하....😢😢😢😢😢

Eyes0n committed:
Modify :peer review 반영 a태그를 Link태그로 전환

yeonuk-hwang commented on src/Pages/hyunchanpark/Login/Login.scss:
```diff
@@ -0,0 +1,160 @@
+            line-height: 51px;
+            margin: 22px auto 12px;
+            font-family: 'Lobster', cursive;
+            @include font-set(40px, 100, null);
```
👍

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Comment.js:
```diff
@@ -0,0 +1,44 @@
+            <span>{comment.content}</span>
+          </div>
+          <button>
+            {comment.userName === 'hyunchan' ? (
+              <i
+                className="far fa-trash-alt"
+                id={comment.id}
+                onClick={e => onRemoveComment(e, feedId)}
+              />
+            ) : null}
```
이 부분은 삼항연산자가 아니라 && 연산자로도 표현 가능해보입니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Comment.js:
```diff
@@ -0,0 +1,44 @@
+
+            <i
+              className={`${
+                // comment.isLiked
```
불필요한 주석은 삭제해주세요

Eyes0n replied:
댓글 좋아요 부분이 기존 데이터가 단일 사용자에 대해  boolea값으로 존재했었는데 
저는 하나의 댓글에 여러 사용자가 좋아요를 할 수 있으니깐 배열로 바꿔서 좋아요/취소 기능을 구현했습니다.
보통 sns와 같은 좋아요 기능 구현할 때 데이터 테이블에서 가져오는 값 형태가 단일 사용자를 위한 값으로 boolean을 주는게 일반적인가요?

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Comment.js:
```diff
@@ -0,0 +1,44 @@
+                  ? 'fas'
+                  : 'far'
+              } fa-heart`}
+              id={comment.id}
```
id selector는 사용을 지양해주세요!
id는 전체 document에서 하나만 존재해야합니다!
그런데 React의 Component는 일반적으로 여러군데서 재사용이 가능한 목적으로 설계하고 있습니다.
만약 이 컴포넌트가 재사용된다면 전체 도큐먼트에서 같은 id를 가진 element들이 중복으로 발생되어서 id가 유일하다고 보장이 안되겠네요
그러한 이유로 인해서 id는 사용을 지양하고있습니다.

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/CommentInput.js:
```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+
+class CommentInput extends Component {
+  constructor(props) {
```
constructor안에서 props를 사용할 것이 아니라면 props는 명시 안해주셔도 될 것 같습니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/CommentInput.js:
```diff
@@ -0,0 +1,76 @@
+  };
+
+  onSumitInputValue = e => {
+    const newComment = this.state.inputValue;
+    const { feedId } = this.props;
+    this.props.onSubmitCommentForm(e, feedId, newComment);
+    this.setState({ inputValue: '' });
```
너무 깔끔하네요 👍

Eyes0n replied:
감사합니다 ㅎㅎ💕💕💕

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/CommentInput.js:
```diff
@@ -0,0 +1,76 @@
+  render() {
+    const { inputValue } = this.state;
+    const { feedId } = this.props;
+    const isSubmitDisable = inputValue && inputValue.length > 0 ? false : true;
```
Login.js에 달아드린 코멘트와 유사한 부분인 것 같습니다

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/CommentList.js:
```diff
@@ -0,0 +1,47 @@
+          </div>
+
+          <div className="other-comments">
+            {commentList &&
+              commentList.map((comment, i) => (
```
이런식으로 commentList안에 아직 배열이 안들어왔을때 array method를 활용할 경우 나는 에러를 피하기 위해서 && 연산자를 사용하는 경우 [optional chaining](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining) 을 사용해주실수도 있습니다.

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Feed.js:
```diff
@@ -0,0 +1,149 @@
+
+    return (
+      <>
+        <article className="feed" id={feed.id}>
```
id selector 사용은 지양해주세요

Eyes0n replied:
불필요한 부분이라 삭제했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Feed.js:
```diff
@@ -0,0 +1,149 @@
+              </div>
+            </div>
+            <button>•••</button>
+          </header>
```
header는 구획상의 윗부분을 표현하기 위한 tag가 아닙니다!
header 태그가 사용되는 케이스 다시 확인해보시면 좋을 것 같습니다.

[참고자료](https://developer.mozilla.org/ko/docs/Web/HTML/Element/header)

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Feed.js:
```diff
@@ -0,0 +1,149 @@
+          </header>
+
+          <div className="feed-image">
+            <img src={feed.feedImgSrc} alt="feedImage" />
```
alt는 대체텍스트로도 표시되는 문구이기때문에 카멜케이스가아닌 일반 영문법을 지켜서 작성해주시면 됩니다.
그리고 img tag의 attribute이기 때문에 이미 image라는 것은 알고 있습니다.
따라서 image라는 단어는 굳이 붙여줄 필요가 없습니다.

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/FeedList.js:
```diff
@@ -0,0 +1,128 @@
+  onSubmitCommentForm = (e, feedId, newComment) => {
+    e.preventDefault();
+
+    if (newComment === '') return;
```
빈 스트링은 falsy값이기때문에 아래와 같이 표현 가능합니다!
truthy falsy값 다시한번 확인해보시고 잘 활용하시면 좋겠습니다.
```suggestion
    if (!newComment) return;
```

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/FeedList.js:
```diff
@@ -0,0 +1,128 @@
+
+    const { feedList } = this.state;
+
+    const nextState = feedList.map(feed =>
```
nextState라는 이름보다는 newFeedList, updatedFeedList 등이 좀 더 명확한 이름이 아닐까요??

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/FeedList.js:
```diff
@@ -0,0 +1,128 @@
+          }
+        : feed
+    );
+    console.log('new', nextState);
```
test 마친 콘솔은 삭제해주세요

Eyes0n replied:
네 리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/FeedList.js:
```diff
@@ -0,0 +1,128 @@
+        <div className="feeds">
+          {feedList.map((feed, i) => (
+            <Feed
+              key={i}
```
index로 key를 활용하기 보다 feedData에 id값이 있다면 그걸 활용하시는게 좋아보입니다!

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Pages/hyunchanpark/Main/Main.js:
```diff
@@ -0,0 +1,246 @@
+                            className="link-background"
+                            style={{
+                              background:
+                                'url("/images/hyunchanpark/images/insta-sprite.png) no-repeat -306px -66px',
```
inline style은 지양해주세요
- 우선순위가 제일 높게 적용되기 때문에 추후 유지보수가 힘들어지고
- css파일에서 스타일을 확인할텐데 js에 스타일이 흩어져있으면 확인하기가 힘들어집니다.

Eyes0n replied:
리뷰 반영했습니다.

yeonuk-hwang commented on src/Routes.js:
```diff
@@ -1,26 +1,26 @@
+          <Route exact path="/" component={LoginChan} />
+          <Route path="/main" component={MainChan} />
+          {/* <Route exact path="/main-sanghun" component={LoginHun} />
+          <Route exact path="/main-sanghun" component={MainHun} /> */}
```
팀원들의 코드를 주석처리한 상태로 마스터에 머지가 된다면 문제가 발생하겠죠?
PR은 마스터에 머지될 준비가 되어있어야 합니다.
팀원 코드 주석 해제해주세요!

Eyes0n replied:
npm start하면 브라우저에 해당 파일이 비어있기 때문에 불러올 수 없어 발생하는 오류로인해 주석 처리를 해준것인데 merge를 위해 풀어주는게 맞나요?

yeonuk-hwang reviewed (commented):
코드 깔끔하게 잘 작성해주셨네요 수고하셨습니다!
리뷰 확인 & 반영해주세요!
PR 제목은 `[이름] 작업내용 요약`
으로 맞춰주세요!

Eyes0n committed:
Modify : 리뷰 피드백 반영
